trieste/acquisition/function.py

```diff
@@ -676,3 +676,243 @@ def batch_ei(at: TensorType) -> TensorType:
             return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
 
         return batch_ei
+
+
+class GreedyAcquisitionFunctionBuilder(ABC):
```

Comment:
not sure this is for this PR but this builder could also be used in the asynchronous case I believe? (i.e. we would choose the next experiment while B-1 are still running) In that case maybe Greedy is not the right term?